.github/workflows/e2e.yml

```diff
@@ -7,6 +7,10 @@ on:
   push:
     branches: 'main'
 
+concurrency:
+  group: e2e-${{ github.head_ref || github.run_id }}
```

Comment:
Can you use `${{ github.workflow }}` instead of hard-coding `e2e` here? Same to all other workflows.

Please double-check if `github.workflow` evals to "e2e" (or to "e2e.yml") though.